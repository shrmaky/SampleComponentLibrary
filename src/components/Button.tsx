import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

interface ButtonProps {
    title: string;
    onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({ title, onPress }) => {
    return (
        <TouchableOpacity style={styles.button} onPress={onPress}>
            <Text style={styles.text}>{title}</Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    button: {
        backgroundColor: '#007AFF',
        borderRadius: 8,
        padding: 16,
    },
    text: {
        color: '#FFFFFF',
        fontSize: 16,
        fontWeight: 'bold',
        textAlign: 'center',
    },
});

export default Button;