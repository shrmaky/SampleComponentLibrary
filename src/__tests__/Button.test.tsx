import React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<Button title="Press me" onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});