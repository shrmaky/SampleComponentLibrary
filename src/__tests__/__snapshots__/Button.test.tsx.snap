// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Press me
  </Text>
</View>
`;
